yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,125 @@
+          cache. To read keyword from cache the target should be 0 and to read
+          from hardware the target is 1.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Path of the FRU whose keyword needs to be read.
+          - name: Keyword
+            type: struct[string,string]
+            description: >
+                It is a tuple of record and keyword to be read.
+          - name: target
+            type: size
+            description: >
+                Target where the keyword to be read from cache/hardware.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                value of the corresponding keyword.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - com.ibm.VPD.Error.PathNotFound
+          - com.ibm.VPD.Error.RecordNotFound
+
+    - name: GetHardwarePath
+      description: >
+          An api to get hardware path for the corresponding FRU DBus object
+          path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                DBus object path of the FRU whose hardware path needs to be
+                collected.
+      returns:
+          - name: Path
+            type: string
+            description: >
+                Hardware path of the corresponding FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetExpandedLocationCode
+      description: >
+          An api to get expanded location code for the corresponding FRU DBus
+          path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                DBus path of the FRU whose location code needs to be collected.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in expanded format.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: deleteFRUVPD
+      description: >
+          An api to delete VPD of a given FRU and set its Present property as
+          false on DBus.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                DBus path of the FRU whose VPD needs to be removed.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: CollectFRUVPD
+      description: >
+          API can be used to perform VPD collection for the given FRU. The FRU
+          should have concurrentlyMaintainable flag set to true in VPD JSON to
+          execute this action.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Dbus path of the FRU whose VPD needs to be collected.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
```

Comment:
can you remove Invalid argument and add com.ibm.VPD.Error.PathNotFound

Reply:
Done

Reply:
invalid argument is valid as well as other error types needs to be set if any. think on this more

Reply:
updated